### PR TITLE
Created Bike Index module, added tests that pass

### PR DESCRIPTION
**Pull request by sethherr**
I added a [Bike Index](https://bikeindex.org/documentation) module. 

I couldn't get the demo to work locally because the redirect url isn't valid, since it's a local file not a URL.

What should I do next? What can I do to make merging/testing this easier?

### CODE REVIEW

**Comment by MrSwitch** [comment]
Thanks Seth, I love the site!
I'll check your code out ASAP

**Comment by MrSwitch** [comment]
Hi Seth

Your module looks really good.

Could you rephrase your comment in `demos/bikeindex.html` as i dont understand that.
> // You can redirect to a demo doc on the index via urn:ietf:wg:oauth:2.0:oob

I presume you want to use "urn:ietf:wg:oauth:2.0:oob" for the redirect_uri. I haven't enabled this yet in HelloJS. Currently it will treat this as a relative URL as i'm sure you've realised.

FYI: demo/bikeindex does not need to define an oauth_proxy since your API supports implicit authentication.

The "me" endpoint would be great if it showed the user `name`. Can we enable that? I felt like creating a PR for this but i'm on holiday and it might annoy my other half. :)

btw: has anyone created a heatmap for stolen bikes using your API?

**Comment by sethherr** [comment]
Awesome Andrew! Thanks for looking through this.

I pushed an update to include name and remove the confusing comment.

Can/should I include the non standard scopes as options? Currently [they're commented out](https://github.com/sethherr/hello.js/blob/master/src/modules/bikeindex.js#L52)

No one has created a live heatmap thing, HotPads [made one for specific cities](http://hotpads.com/blog/2014/11/preventing-bike-thievery/) a while ago. HotPads actually used (and we'd recommend using) BikeWise, since [its API](https://bikewise.org/documentation) outputs geojson and includes all the stolen bikes from the Index.

**Comment by MrSwitch** [comment]
I've made a change to the defaults https://github.com/MrSwitch/hello.js/commit/8f0cf57475df20cc0c1793ad6f77309ebff07a6a

Please merge via `git pull https://github.com/sethherr/hello.js.git sethherr-master`
( this includes the latest changes from master to make this work with your api scope parameters https://github.com/MrSwitch/hello.js/commit/57cfe2087190b37ccfc4fb8de9f5c4bf50532be8)

**Comment by MrSwitch** [comment]
> Can/should I include the non standard scopes as options

No, they are not generic enough. But keep them in their under comments. If might be better to update `demos/bikeindex.html`

**Comment by sethherr** [comment]
Is there anything else I can/should do? I updated the demo page with a slightly better description of what happens when you click the button.

What needs to be done for this to be merged in?

**Comment by MrSwitch** [comment]
Merged 602b2bb6e into master - better late than never.

I am having trouble updating my app setting on https://bikeindex.org/oauth/applications/19, its giving me large "500" error when i try to amend the permitted redirect_uris. If you get a chance can you update my domain names which should be "adodson.com" instead of "adodson"

Then http://adodson.com/hello.js/demos/bikeindex.html should work!

**Comment by sethherr** [comment]
Bah, sorry for the slow reply. Thanks!
